ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpDataHelper.java

```diff
@@ -0,0 +1,589 @@
+            BlobData blobData = MessageFormatRecord.deserializeBlob(streamlog);
+            blobDataOutput = "Blob - size " + blobData.getSize();
+          } else {
+            boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(streamlog);
+            isDeleted = true;
+            deleteMsg = "delete change " + deleteFlag;
+          }
+          lastBlobFailed = false;
+          if (!isDeleted) {
+            if (filter) {
+              if (blobs.contains(id.getID())) {
+                logOutput(
+                    messageheader + "\n " + blobId + "\n" + blobProperty + "\n" + usermetadata + "\n" + blobDataOutput);
+                updateBlobIdToLogRecordMap(blobIdToLogRecord, id.getID(), currentOffset, !isDeleted, isExpired);
+              }
+            } else {
+              logOutput(
+                  messageheader + "\n " + blobId + "\n" + blobProperty + "\n" + usermetadata + "\n" + blobDataOutput);
+              updateBlobIdToLogRecordMap(blobIdToLogRecord, id.getID(), currentOffset, !isDeleted, isExpired);
+            }
+          } else {
+            if (filter) {
+              if (blobs.contains(id.getID())) {
+                logOutput(messageheader + "\n " + blobId + "\n" + deleteMsg);
+                updateBlobIdToLogRecordMap(blobIdToLogRecord, id.getID(), currentOffset, !isDeleted, isExpired);
+              }
+            } else {
+              logOutput(messageheader + "\n " + blobId + "\n" + deleteMsg);
+              updateBlobIdToLogRecordMap(blobIdToLogRecord, id.getID(), currentOffset, !isDeleted, isExpired);
+            }
+          }
+          currentOffset += (header.getMessageSize() + buffer.capacity() + id.sizeInBytes());
+        } else {
+          if (!lastBlobFailed) {
+            logOutput("Header Version not supported. Thrown at reading a msg starting at " + tempCurrentOffset);
+            lastBlobFailed = true;
+          }
+          randomAccessFile.seek(++tempCurrentOffset);
+          currentOffset = tempCurrentOffset;
+        }
+      } catch (IllegalArgumentException e) {
+        logOutput("Illegal arg exception thrown at  " + randomAccessFile.getChannel().position() + ", " +
+            "while reading blob starting at offset " + tempCurrentOffset + " with " + messageheader + blobId
+            + blobProperty + usermetadata + blobDataOutput + " exception: " + e);
+        randomAccessFile.seek(++tempCurrentOffset);
+        currentOffset = tempCurrentOffset;
+      } catch (MessageFormatException e) {
+        logOutput("MessageFormat exception thrown at  " + randomAccessFile.getChannel().position() +
+            " while reading blob starting at offset " + tempCurrentOffset + " with " + messageheader + blobId
+            + blobProperty + usermetadata + blobDataOutput + " exception: " + e);
+        randomAccessFile.seek(++tempCurrentOffset);
+        currentOffset = tempCurrentOffset;
+      } catch (EOFException e) {
+        e.printStackTrace();
```

Comment:
if the exception is going to be thrown, is it necessary to print the stack trace?